ChainBridge documentation (#24)
0xPolygon/polygon-sdk-docs: docs/chainbridge/setup/setup-erc721-transfer.md

---
id: setup-erc721-transfer
title: ERC721 NFT Transfer
---

> This section guides you a way how to set up ERC721 bridge and send NFT to the other chain.

Reviewer: ```suggestion
This section guides you through setting up an ERC721 bridge and sending NFTs between blockchain networks.
```

Author: Fixed at 5756c760532fdfa7da42c47aad4dc893d1e4074e

## Step 1: Register resource ID

> You will need to register resource ID for ERC721 to Bridge contracts on the both chains at first.

Reviewer: ```suggestion
You will first need to register the resource ID for the ERC721 token in the Bridge contracts on both chains.
```

Author: Fixed at 5756c760532fdfa7da42c47aad4dc893d1e4074e

  --privateKey [MINTER_ROLE_ACCOUNT] \
  --erc721Address "[ERC721_CONTRACT_ADDRESS]" \
  --id 0x50
```

To check the NFT owner, you can use `cb-sol-cli erc721 owner`

```bash
# Check the current owner of NFT
$ cb-sol-cli erc721 owner \
  --url https://rpc-mumbai.matic.today \
  --privateKey [PRIVATE_KEY] \
  --erc721Address "[ERC721_CONTRACT_ADDRESS]" \
  --id 0x50
```

Then, you will approve a transfer of the NFT by ERC721 Handler

```bash
# Approve transfer of the NFT 0x50 by ERC721 Handler
$ cb-sol-cli erc721 approve \
  --url https://rpc-mumbai.matic.today \
  --privateKey [PRIVATE_KEY] \
  --erc721Address "[ERC721_CONTRACT_ADDRESS]" \
  --recipient "[ERC721_HANDLER_CONTRACT_ADDRESS]" \
  --id 0x50
```

Finally, you will start transfer

```bash
# Start transfer from Polygon PoS to Polygon SDK chain
$ cb-sol-cli erc721 deposit \
  --url https://rpc-mumbai.matic.today \
  --privateKey [PRIVATE_KEY] \
  --bridge "[BRIDGE_CONTRACT_ADDRESS]" \
  --resourceId "0x000000000000000000000000000000e389d61c11e5fe32ec1735b3cd38c69501" \
  --id 0x50 \
  # ChainID of Polygon SDK chain
  --dest 100 \
  --recipient "[RECIPIENT_ADDRESS_IN_POLYGON_SDK_CHAIN]"
```

> Relayer would get the event and vote for the proposal. It executes a transaction to send NFT to the recipient account in Polygon SDK chain after the required number of votes are submitted. 

Reviewer: ```suggestion
Relayer will get the event and vote for the proposal. It executes a transaction to send NFTs to the recipient account in Polygon SDK chain after the required number of votes are submitted. 
```

Author: Fixed at 5756c760532fdfa7da42c47aad4dc893d1e4074e

```bash
INFO[11-19|09:07:50] Handling nonfungible deposit event       chain=mumbai
INFO[11-19|09:07:50] Attempting to resolve message            chain=polygon-sdk type=NonFungibleTransfer src=99 dst=100 nonce=2 rId=000000000000000000000000000000e389d61c11e5fe32ec1735b3cd38c69501
INFO[11-19|09:07:50] Creating erc721 proposal                 chain=polygon-sdk src=99 nonce=2
INFO[11-19|09:07:50] Watching for finalization event          chain=polygon-sdk src=99 nonce=2
INFO[11-19|09:07:50] Submitted proposal vote                  chain=polygon-sdk tx=0x58a22d84a08269ad2e8d52d8dc038621f1a21109d11c7b6e0d32d5bf21ea8505 src=99 depositNonce=2 gasPrice=1
INFO[11-19|09:08:15] Submitted proposal execution             chain=polygon-sdk tx=0x57419844881a07531e31667c609421662d94d21d0709e64fb728138309267e68 src=99 dst=100 nonce=2 gasPrice=3
```

> You can check the owner in Polygon SDK after execution was completed.

Reviewer: ```suggestion
You can check the owner of the NFT on the Polygon SDK network after the execution is completed.
```

Author: Fixed at 5756c760532fdfa7da42c47aad4dc893d1e4074e
